posydon/popsyn/synthetic_population.py

```diff
@@ -62,38 +68,61 @@ def __init__(self, path_to_ini, path_to_data=None, verbose=False, MODEL={}):
-
-            self.binary_populations = []
-            for met in self.metallicity[::-1]:
-                self.ini_kw = binarypop_kwargs_from_ini(path_to_ini)
-                self.ini_kw['metallicity'] = met
-                self.ini_kw['temp_directory'] = self.create_met_prefix(met) + self.ini_kw['temp_directory']
-                self.binary_populations.append(BinaryPopulation(**self.ini_kw))
-
-        if path_to_data is None:
-            return
-        elif (isinstance(path_to_data, list) and '.h5' in path_to_data[0]) or ('.h5' in path_to_data):
-            self.path_to_data = path_to_data
+            self.binary_populations = None
+        
+    def create_binary_populations(self):
+        """Create a list of BinaryPopulation objects."""
+        self.binary_populations = []
+        ini_kw = self.synthetic_pop_params.copy()
+        for met in self.metallicity[::-1]:
+            ini_kw['metallicity'] = met
+            ini_kw['temp_directory'] = self.create_met_prefix(met) + self.synthetic_pop_params['temp_directory']
+            self.binary_populations.append(BinaryPopulation(**ini_kw))
 
     def get_ini_kw(self):
-        return self.ini_kw
+        return self.synthetic_pop_params.copy()
 
     def evolve(self):
         """Evolve population(s) at given Z(s)."""
+        if self.binary_populations is None:
+            self.create_binary_populations()
         while self.binary_populations:
             pop =  self.binary_populations.pop()
             print( f'Z={pop.kwargs["metallicity"]:.2e} Z_sun' )
             pop.evolve()
-            met_prefix = f'{pop.kwargs["metallicity"]:.2e}_Zsun_'
-            pop.save( met_prefix + 'population.h5' )
             del pop
 
+
+    def merge_parallel_runs(self, path_to_batches):
+        """
+        Merge the folder or list of folders into a single file per metallicity.
+
+        Parameters
+        ----------
+        path_to_batches : str or list of str
+            Path to the folder(s) containing the batch folders.
+        """
+        
+        if isinstance(path_to_batches, str):
+            path_to_batches = [path_to_batches]
+        # check if path_to_batches is the same length as the number of metallicities
+        if len(path_to_batches) != len(self.metallicity):
+            raise ValueError('The number of metallicity and batch directories do not match!')
+
+        for met, path_to_batch in zip(self.metallicity, path_to_batches):
+            met_prefix = self.create_met_prefix(met)
+            tmp_files = [os.path.join(path_to_batch, f)     \
+                         for f in os.listdir(path_to_batch) \
+                            if os.path.isfile(os.path.join(path_to_batch, f))]
+            
+            BinaryPopulation(**self.get_ini_kw()).combine_saved_files(met_prefix+ 'population.h5', tmp_files)
+            print(f'Population at Z={met:.2e} Z_sun successfully merged!')
+            os.rmdir(path_to_batch)
```

Comment:
This might fail if the directory is not empty. The code only removed the files, which where successfully combined into the new file. The could be other files or maybe even some subdirectories left in there.

Reply:
There shouldn't be any files left if correctly finished, but I've created a check to make sure that the folder is empty before it tries to remove it. 
```suggestion
            if len(os.listdir(path_to_batch)) == 0:
```